pylsp/plugins/jedi_completion.py

```diff
@@ -62,36 +62,34 @@ def pylsp_completions(config, document, position):
         SNIPPET_RESOLVER.cached_modules = modules_to_cache_for
 
     include_params = snippet_support and should_include_params and use_snippets(document, position)
-    include_class_objects = snippet_support and should_include_class_objects and use_snippets(document, position)
-
-    ready_completions = [
-        _format_completion(
-            c,
-            include_params,
-            resolve=resolve_eagerly,
-            resolve_label_or_snippet=(i < max_to_resolve)
-        )
-        for i, c in enumerate(completions)
-    ]
-
-    # TODO split up once other improvements are merged
-    if include_class_objects:
-        for i, c in enumerate(completions):
-            if c.type == 'class':
-                completion_dict = _format_completion(
+    include_class_objects = (
+        snippet_support and should_include_class_objects and use_snippets(document, position)
+    )
+
+    ready_completions = []
+    for i, c in enumerate(completions):
+
+        ready_completions.append(
+            {
+                **_format_completion(
                     c,
-                    False,
+                    include_params,
                     resolve=resolve_eagerly,
-                    resolve_label_or_snippet=(i < max_to_resolve)
-                )
-                completion_dict['kind'] = lsp.CompletionItemKind.TypeParameter
-                completion_dict['label'] += ' object'
-                ready_completions.append(completion_dict)
+                    resolve_label_or_snippet=(i < max_to_resolve),
+                ),
+                'data': {'doc_uri': document.uri},
+            }
+        )
+
+        if c.type == 'class' and include_class_objects:
```

Comment:
The proposed change will perform thousands of expensive string comparisons; please switch the order to short-circuit early on cheap bool check:

```suggestion
        if include_class_objects and c.type == 'class':
```

Still, even after switching order, the proposed code will perform thousands of comparisons where the current one would do just one - and it will affect every user whether they have `include_class_objects` enabled or not.

Reply:
Ah I understand - I'll remove for this PR